### ISSUE TITLE
cmd && othercmd throws an exception

### ISSUE THREAD

**Issue by OneOfOne** [xonsh/xonsh#2380]
```
➤ ls &&Exception in thread Thread-35:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.6/site-packages/prompt_toolkit/interface.py", line 860, in run
    completions = list(buffer.completer.get_completions(document, complete_event))
  File "/usr/lib/python3.6/site-packages/xonsh/ptk/completer.py", line 49, in get_completions
    self.ctx)
  File "/usr/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 124, in complete
    out = func(prefix, line, begidx, endidx, ctx)
  File "/usr/lib/python3.6/site-packages/xonsh/completers/__amalgam__.py", line 831, in complete_skipper
    ctx)
  File "/usr/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 124, in complete
    out = func(prefix, line, begidx, endidx, ctx)
  File "/usr/lib/python3.6/site-packages/xonsh/completers/__amalgam__.py", line 1427, in complete_base
    complete_command(prefix, line, start, end, ctx))
TypeError: unsupported operand type(s) for |: 'tuple' and 'set'
```

```
➤ y -Q | egrep 'toolkit|xon'
community/python-prompt_toolkit 1.0.14-1
local/xonsh 0.5.9-1
```

Let me know if you need anything else.

**Comment by gforsyth** [comment]
Thanks for reporting @OneOfOne -- can you confirm that you get this exception when you hit tab?

**Comment by OneOfOne** [comment]
I get it by typing the second `&`, before hitting tab/space.

**Comment by gforsyth** [comment]
Ok, that's much weirder. Can you please post the output of `xonfig` and also your `.xonshrc`?

**Comment by OneOfOne** [comment]
It's a mess right now but https://gist.github.com/OneOfOne/97b03541ddee193407c9445c2d77689a

```
┏━oneofone@Ava ❨~❩
┗━━➤ xonfig 
+------------------+-----------------+
| xonsh            | 0.5.9           |
| Python           | 3.6.1           |
| PLY              | 3.9             |
| have readline    | True            |
| prompt toolkit   | 1.0.14          |
| shell type       | prompt_toolkit  |
| pygments         | 2.2.0           |
| on posix         | True            |
| on linux         | True            |
| distro           | arch            |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

**Comment by gforsyth** [comment]
Try disabling the update on keypress envvar and see if it still throws that
error.

**Comment by OneOfOne** [comment]
That actually fixed it @gforsyth.

**Comment by gforsyth** [comment]
Thanks, @OneOfOne -- this is definitely a bug in the completer framework. It was getting triggered by the update on keypress envvar, but it also happens if you press tab after the `&&`

**Comment by OneOfOne** [comment]
Let me know if you need me to test anything.

**Comment by nh2** [comment]
I can confirm this happening.

**Comment by gerardlouw** [comment]
This happens when typing `||` as well.

I'm working around this bug by using `and` and `or` instead of `&&` and `||`. I guess that's more Pythonic anyways.

**Comment by scopatz** [comment]
@gerardlouw - can you post the output of `xonfig`? I wonder if this only happens on PTKv1.

**Comment by laloch** [comment]
I can confirm that it also happens with PTK2. The trick is to press the <kbd>TAB</kbd> key immediately after typing the second `|` or `&` character (without the trailing space). Un-amalgamated backtrace follows:
```
str@str ~ $ xonfig                                                                                                                                                                                                                                                          
+------------------+---------------------+
| xonsh            | 0.8.8               |
| Git SHA          | 67db1fa8            |
| Commit Date      | Oct 3 02:58:28 2018 |
| Python           | 3.7.1               |
| PLY              | 3.11                |
| have readline    | True                |
| prompt toolkit   | 2.0.7               |
| shell type       | prompt_toolkit2     |
| pygments         | 2.3.1               |
| on posix         | True                |
| on linux         | True                |
| distro           | arch                |
| on darwin        | False               |
| on windows       | False               |
| on cygwin        | False               |
| on msys2         | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
+------------------+---------------------+

str@str ~ $ ls &&
Unhandled exception in event loop:
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 90, in step_next
    new_f = coroutine.throw(exc)
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/buffer.py", line 1648, in new_coroutine
    yield From(coroutine(*a, **kw))
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 90, in step_next
    new_f = coroutine.throw(exc)
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/buffer.py", line 1500, in async_completer
    cancel=lambda: not proceed()))
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 86, in step_next
    new_f = coroutine.send(None)
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/eventloop/async_generator.py", line 114, in consume_async_generator
    item = iterator.send(send)
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/completion/base.py", line 166, in get_completions_async
    for item in self.get_completions(document, complete_event):
  File "/home/str/projects/xonsh/xonsh/ptk2/completer.py", line 45, in get_completions
    prefix, line_ex, begidx + expand_offset, endidx + expand_offset, self.ctx
  File "/home/str/projects/xonsh/xonsh/completer.py", line 36, in complete
    out = func(prefix, line, begidx, endidx, ctx)
  File "/home/str/projects/xonsh/xonsh/completers/commands.py", line 59, in complete_skipper
    cmd, " ".join(parts[skip_part_num:]), start - skip_len, end - skip_len, ctx
  File "/home/str/projects/xonsh/xonsh/completer.py", line 36, in complete
    out = func(prefix, line, begidx, endidx, ctx)
  File "/home/str/projects/xonsh/xonsh/completers/base.py", line 16, in complete_base
    prefix, line, start, end, ctx

Exception unsupported operand type(s) for |: 'tuple' and 'set'
```

**Comment by scopatz** [comment]
Potential fix in #3002
